src/components/checkboxTree/utils.ts

```diff
@@ -0,0 +1,116 @@
+import { Composition } from "./types";
+
+export const preProcessTree = (
+  tree: Composition[],
+  seperator: string
+): Composition[] => {
+  const treeClone = JSON.parse(JSON.stringify(tree));
+  const mappedUids = getRelations(tree, seperator);
+
+  return attachRelations(treeClone, mappedUids, seperator);
+};
+
+export const getChildrenUids = (composition: Composition): string[] => {
+  const uids: string[] = [];
+
+  if (!composition.children) {
+    return [];
+  }
+
+  const recursive = (children: Composition[]) => {
```

Comment:
maybe `recurse` so it's a verb, like other functions

Reply:
Changed to `recursivelyIterateChildren` to provide some context.